fncas/fncas_optimize.h

```diff
@@ -0,0 +1,266 @@
+/*******************************************************************************
+The MIT License (MIT)
+
+Copyright (c) 2015 Dmitry "Dima" Korolev <dmitry.korolev@gmail.com>
+          (c) 2015 Maxim Zhurovich <zhurovich@gmail.com>
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+*******************************************************************************/
+
+#ifndef FNCAS_OPTIMIZE_H
+#define FNCAS_OPTIMIZE_H
+
+#include <algorithm>
+#include <numeric>
+#include <string>
+
+#include "fncas_mathutil.h"
+
+namespace fncas {
+
+struct OptimizationResult {
+  double value;
+  std::vector<double> point;
+};
+
+class OptimizerParameters {
+public:
```

Comment:
Hey, do you have `clang-format` installed (on Ubuntu, I haven't tested it on Mac, ironically)?

If you do `make indent` would make the code comply with Google C++ Style Guide in many ways, like one space before `public:`/`protected:` / `private:`.